sys/include/vfs.h

```diff
@@ -65,6 +65,46 @@ extern "C" {
 #define VFS_MOUNT_POINT_LEN (8)
 #endif
 
+#ifndef VFS_DIR_BUFFER_SIZE
+/**
+ * @brief Size of buffer space in vfs_DIR
+ *
+ * This space is needed to avoid dynamic memory allocations for some file
+ * systems where a single pointer is not enough space for its directory stream
+ * state, e.g. SPIFFS.
+ *
+ * Guidelines:
+ *
+ * SPIFFS requires a sizeof(spiffs_DIR) (6-16 bytes, depending on target
+ * platform and configuration) buffer for its DIR struct.
+ *
+ * @attention File system developers: If your file system requires a buffer for
+ * DIR streams that is larger than a single pointer or @c int variable, ensure
+ * that you have a preprocessor check in your header file (so that it is
+ * impossible to attempt to mount the file system without running into a
+ * compiler error):
+ *
+ * @attention @code
+ * #if VFS_DIR_BUFFER_SIZE < (sizeof(my_DIR))
```

Comment:
@gebart I'm not sure this can actually work. Is it really possible to use sizeof in a preprocessor expression ?